camayoc/tests/qpc/cli/test_openshift.py

```diff
@@ -77,6 +78,50 @@ def validate_openshift_report(cluster, details, deployments):
     assert set(cluster_nodes) == set(seen_nodes), "The number of expected nodes diverged"
 
 
+def validate_node_metrics(cluster, details):
+    node_metrics = details["sources"][0]["facts"][-1]["node_metrics"]
+    for metric in node_metrics:
+        package = ast.literal_eval(metric["package"])
+        assert package >= 0
+        assert metric["label_node_hyperthread_enabled"] in ("true", "false")
+
+
+def validate_operators(cluster, details):
+    expected_operators = cluster["operators"]
+    detected_operators = [op["name"] for op in details["sources"][0]["facts"][-1]["operators"]]
+    for operator in expected_operators:
+        assert operator in detected_operators, f"The operator '{operator}' was not detected."
+
+
+def validate_openshift_with_acm(cluster, details):
+    EXPECTED_METRICS_KEYS = (
+        "vendor",
+        "cloud",
+        "version",
+        "managed_cluster_id",
+        "available",
+        "core_worker",
+        "socket_worker",
+        "created_via",
+    )
+    cluster_id = cluster["cluster_id"]
+    # The last 'facts' in the list will contains cluster, operators and acm_metrics information
+    acm_metrics = details["sources"][0]["facts"][-1]["acm_metrics"]
+    assert (
+        len(acm_metrics) > 1
+    ), "Cluster has ACM and no metrics were detected (expected at least one)"
```

Comment:
Message says at least one metric was expected, but assertion will pass only if there are 2 or more.

Reply:
My test case is based upon two clusters, the first cluster has ACM and the second cluster is registered to this first cluster (no ACM). Well, it's good enough to assume `> 0` here (at least one metric, ACM is installed = one cluster).

```
(Pdb) acm_metrics[0]
{'vendor': 'OpenShift', 'cloud': 'Other', 'version': '4.13.4', 'managed_cluster_id': '...', 'available': True, 'core_worker': '12', 'socket_worker': '12', 'created_via': 'other'}
(Pdb) acm_metrics[1]
{'vendor': 'OpenShift', 'cloud': 'Other', 'version': '4.12.25', 'managed_cluster_id': '...', 'available': True, 'core_worker': '12', 'socket_worker': '12', 'created_via': 'other'}
(Pdb) acm_metrics[2]
*** IndexError: list index out of range
```

Reply:
Either way is fine for me - I'm not in position to say which one is more correct than the other. I was only pointing out that message said something else than assertion checked.

Now message and assertion align, so it's all good.